enums.go

```diff
@@ -255,127 +255,155 @@ func (d Datatype) MakeSlice(numElements uint64) (interface{}, unsafe.Pointer, er
+			tmpslice := (*[1 << 46]C.uint8_t)(cvalue)[:valueNum:valueNum]
 			for i, s := range tmpslice {
 				tmpValue[i] = uint8(s)
 			}
-			value = tmpValue
-		} else {
-			value = *(*uint8)(unsafe.Pointer(cvalue))
+			return tmpValue, nil
 		}
+		return *(*uint8)(cvalue), nil
 	case TILEDB_UINT16:
+		if cvalue == nil {
+			return uint16(0), nil
+		}
 		if valueNum > 1 {
 			tmpValue := make([]uint16, valueNum)
-			tmpslice := (*[1 << 46]C.uint16_t)(unsafe.Pointer(cvalue))[:valueNum:valueNum]
+			tmpslice := (*[1 << 46]C.uint16_t)(cvalue)[:valueNum:valueNum]
 			for i, s := range tmpslice {
 				tmpValue[i] = uint16(s)
 			}
-			value = tmpValue
-		} else {
-			value = *(*uint16)(unsafe.Pointer(cvalue))
+			return tmpValue, nil
 		}
+		return *(*uint16)(cvalue), nil
 	case TILEDB_UINT32:
+		if cvalue == nil {
+			return uint32(0), nil
+		}
 		if valueNum > 1 {
 			tmpValue := make([]uint32, valueNum)
-			tmpslice := (*[1 << 46]C.uint32_t)(unsafe.Pointer(cvalue))[:valueNum:valueNum]
+			tmpslice := (*[1 << 46]C.uint32_t)(cvalue)[:valueNum:valueNum]
 			for i, s := range tmpslice {
 				tmpValue[i] = uint32(s)
 			}
-			value = tmpValue
-		} else {
-			value = *(*uint32)(unsafe.Pointer(cvalue))
+			return tmpValue, nil
 		}
+		return *(*uint32)(cvalue), nil
 	case TILEDB_UINT64:
+		if cvalue == nil {
+			return uint64(0), nil
+		}
 		if valueNum > 1 {
 			tmpValue := make([]uint64, valueNum)
-			tmpslice := (*[1 << 46]C.uint64_t)(unsafe.Pointer(cvalue))[:valueNum:valueNum]
+			tmpslice := (*[1 << 46]C.uint64_t)(cvalue)[:valueNum:valueNum]
 			for i, s := range tmpslice {
 				tmpValue[i] = uint64(s)
 			}
-			value = tmpValue
-		} else {
-			value = *(*uint64)(unsafe.Pointer(cvalue))
+			return tmpValue, nil
 		}
+		return *(*uint64)(cvalue), nil
 	case TILEDB_FLOAT32:
+		if cvalue == nil {
+			return float32(0), nil
+		}
 		if valueNum > 1 {
 			tmpValue := make([]float32, valueNum)
-			tmpslice := (*[1 << 46]C.float)(unsafe.Pointer(cvalue))[:valueNum:valueNum]
+			tmpslice := (*[1 << 46]C.float)(cvalue)[:valueNum:valueNum]
 			for i, s := range tmpslice {
 				tmpValue[i] = float32(s)
 			}
-			value = tmpValue
-		} else {
-			value = *(*float32)(unsafe.Pointer(cvalue))
+			return tmpValue, nil
 		}
+		return *(*float32)(cvalue), nil
 	case TILEDB_FLOAT64:
+		if cvalue == nil {
+			return float64(0), nil
+		}
 		if valueNum > 1 {
 			tmpValue := make([]float64, valueNum)
-			tmpslice := (*[1 << 46]C.double)(unsafe.Pointer(cvalue))[:valueNum:valueNum]
+			tmpslice := (*[1 << 46]C.double)(cvalue)[:valueNum:valueNum]
 			for i, s := range tmpslice {
 				tmpValue[i] = float64(s)
 			}
-			value = tmpValue
-		} else {
-			value = *(*float64)(unsafe.Pointer(cvalue))
+			return tmpValue, nil
 		}
+		return *(*float64)(cvalue), nil
 	case TILEDB_CHAR:
-		tmpslice := (*[1 << 46]C.char)(unsafe.Pointer(cvalue))[:valueNum:valueNum]
-		value = C.GoString(&tmpslice[0])[0:valueNum]
+		if cvalue == nil {
```

Comment:
Need to also check for a length of 0 because a zero-length string could be non-null. Empty string `""` is valid.